### DISCUSSION TITLE
Missing dll for Friction Liquid?

### DISCUSSION THREAD

**Discussion by mikael-onsilva** [Anura3D/Anura3D_OpenSource#151] [Q&A]
Hello everyone,

Am I wrong or is the the dll for the Friction Liquid constitutive model? At the code (ReadMaterialData.FOR, case ESM_FRICTIONAL_LIQUID) doesn't initialize the dll either.
Is that right?

On my tests, I change the parameters in the model (column collapse example, with Friction Liquid material instead of Mohr Coulomb) but the results remain the same, with all the material spreading like water, with no deposition angle.

I've already found out that GiD isn't creating the GOM file correctly, som I'm already editing it manually with the parameters below:
$$MATERIAL_MODEL_LIQUID
frictional_liquid
$$LIQUID_FRICTION_ANGLE
30
$$LIQUID_YOUNG_MODULUS
100000
$$LIQUID_POISSON_RATIO
0.49

Is it really missing the file?

Thanks in advance.

**Comment by murphy-john** [comment]
The Frictional Liquid soil model DLL is not included in the open source code. The material models which are currently provided with the open source version are: Linear Elastic, Mohr Coulomb (Standard), Newtonian Fluid, and Bingham Fluid. The DLL's for other material models must be written by the users.

**Reply by mikael-onsilva** [reply]
Thanks, John.
Is the addition of this or other constitutive models in the pipeline?

Does anyone already have a dll for Frictional Liquid? Or some instructions so I can create one by myself?

Thanks again.

**Reply by murphy-john** [reply]
I'm not aware of any plans to release additional material models. We created the ability to use external soil model so each user could use their own. 

Maybe someone has one you can use. I don't think we have a guide. You could take one of the existing material model codes and use that as a guide.
